app/src/main/java/com/movieapp/repository/MovieRepository.java

```diff
@@ -0,0 +1,130 @@
+package com.movieapp.repository;
+
+import android.app.Application;
+import android.os.AsyncTask;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import com.movieapp.dao.MovieDao;
+import com.movieapp.entity.MovieDB;
+import com.movieapp.model.Movie;
+import com.movieapp.model.TopRatedMoviesResponse;
+import com.movieapp.network.MovieDatabaseAPI;
+import com.movieapp.network.NetworkAPI;
+import com.movieapp.util.MovieDatabase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class MovieRepository {
+    private MovieDao movieDao;
+    private MovieDatabaseAPI movieDatabaseAPI;
+    private MutableLiveData<List<Movie>> movieList;
+    private List<Movie> responseList;
+
+    public MovieRepository(Application application) {
+        MovieDatabase database = MovieDatabase.getInstance(application);
```

Comment:
Please create Application class where you init DB

Reply:
Done